src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/DispatchOperation.cs

```diff
@@ -193,6 +193,8 @@ internal bool IsSessionOpenNotificationEnabled
 
         internal Lazy<AuthorizationPolicy> AuthorizationPolicy { get; set; }
 
+        internal IServiceProvider ServiceProvider { get; set; }
+
```

Comment:
I'm not liking this approach as DispatchOperation is publicly instantiable. If someone creates an instance and replaces the one CoreWCF created, then this can lead to a break. This is only being used to carry the IServiceProvider to OperationInvokerBehavior, but that doesn't need it. You can capture the IServiceProvider from OperationInvokerBehavior.AddBindingParameters from the BindingParameters. It's added to the binding parameters [here](https://github.com/CoreWCF/CoreWCF/blob/main/src/CoreWCF.Primitives/src/CoreWCF/Description/DispatcherBuilder.cs#L172)